Makefile

```diff
@@ -181,17 +181,17 @@ endif
 calico/apiserver: $(BINDIR)/apiserver $(BINDIR)/filecheck
 	rm -rf docker-image/bin
 	mkdir -p docker-image/bin
-	cp $(BINDIR)/apiserver docker-image/bin/
-	cp $(BINDIR)/filecheck docker-image/bin/
-	docker build --pull -t calico/apiserver --file ./docker-image/Dockerfile.$(ARCH) docker-image
+	cp $(BINDIR)/apiserver-$(ARCH) docker-image/bin/
+	cp $(BINDIR)/filecheck-$(ARCH) docker-image/bin/
+	docker build --pull --no-cache --platform linux/$(ARCH) -t calico/apiserver:latest-$(ARCH) --build-arg QEMU_IMAGE=$(CALICO_BUILD) --file ./docker-image/Dockerfile.$(ARCH) docker-image
 
 .PHONY: lint-cache-dir
 lint-cache-dir:
 	mkdir -p $(CURDIR)/.lint-cache
 
-check-boring-ssl: $(BINDIR)/apiserver
+check-boring-ssl: $(BINDIR)/apiserver register
```

Comment:
we only support this on amd64 so no point trying to assert this on non-amd architectures - they will fail.